perfaware/sim86/shared/Sim8086.cpp

```diff
@@ -122,13 +161,14 @@ static void SimulateInstruction(const instruction& Instruction, s16* Registers,
                 } break;
             }
         } break;
-        case Op_add:
-        case Op_sub:
-        case Op_cmp: [[fallthrough]];
+        case Op_add: [[fallthrough]];
+        case Op_sub: [[fallthrough]];
+        case Op_cmp: 
         {
             const instruction_operand& Dest = Instruction.Operands[0];
             const instruction_operand& Source = Instruction.Operands[1];
-            u16 Value = GetSourceValue(Source, Registers);
+            u16 RightOperandValue = GetRightOperandValue(Source, Registers);
+            u16 LeftOperandValue = 0xFFFF;
             u16 Result = 0xFFFF;
```

Comment:
Done! 7af6528f587b219b5192067372b5324d8bdb90a0

Reply:
You defined an `int`, but you use it with `s/u16` values everywhere. So it would be clearer if it were defined as a `u16`.

Reply:
Alright b93859f00d703247a920806aca6c8513c93e9120
Since the other constexpr are used to access array indices, should they be changed to size_t to fit with the theme? 
Or does it only matter here because there are actual operations being done

Reply:
If you want the code to be really robust yeah you'd do that. But it's like I said in the previous PR, it's ok to not do it if you're certain that your array is smaller than the max value of your type, and that it will never get bigger. Some purists would say you should use `size_t` anyway, and there's no issue with that. But you might also choose not to do it, and personally I wouldn't say anything. I just wanted to let you know about the existence of `size_t` last time haha.

Reply:
Cheers!